tests/imgls.pl

```diff
@@ -166,35 +433,129 @@ sub get_image_bytes {
     return $filebytes;
 }
 
-sub get_cmd_output {
-    my $prog = shift;
+sub ls_sort {
+    return if ! (@_ or scalar @_ > 1);
 
-    use English;
-    my ($ret, $pid);
-    die "Can't fork: $!" unless defined($pid = open(KID, "-|"));
-    if ($pid) {           # parent
-        $ret .= $_       while <KID>;
-        close KID;
-        return $ret;
+    # XXX t and r issue and y
+    if ($opts{'t'}) {
+        # descending
+        @_ = sort { _lstat($b)->mtime <=> _lstat($a)->mtime } @_;
+    }
+    # ctime, then lexical within times
+    # does not match macOS' ls(1) cmd, which seems wrong (or at least does not
+    # match the man page) nor does it appear to match any info returned by
+    # stat(1) or mdls.
+    elsif ($opts{'c'}) {
+        @_ = sort { _lstat($a)->ctime <=> _lstat($b)->ctime || $a cmp $b } @_;
+    }
+    elsif ($opts{'u'}) {
+        @_ = sort { _lstat($a)->atime <=> _lstat($b)->atime } @_;
+    }
+    elsif ($opts{'S'}) {
+        @_ = sort { _lstat($b)->size <=> _lstat($a)->size || $a cmp $b } @_;
     }
     else {
-        my @temp     = ($EUID, $EGID);
-        my $orig_uid = $UID;
-        my $orig_gid = $GID;
-        $EUID = $UID;
-        $EGID = $GID;
-        # Drop privileges
-        $UID  = $orig_uid;
-        $GID  = $orig_gid;
-        # Make sure privs are really gone
-        ($EUID, $EGID) = @temp;
-        die "Can't drop privileges"
-            unless $UID == $EUID  && $GID eq $EGID;
-        $ENV{PATH} = "/bin:/usr/bin"; # Minimal PATH.
-    # Consider sanitizing the environment even more.
-        exec $prog, @_,
-            or warn "Failed to run $prog: $!";
-    }
-}
-
-# vim: set expandtab
+        @_ = sort @_;
+    }
+
+    return $opts{'r'} ? reverse @_ : @_;
+}
+
+sub get_F_type {
+    my $st = shift;
+
+    return '/'  if -d $st and ($opts{'p'} or $opts{'F'});
+    return ''   unless $opts{'F'};
+    return '@'  if -l $st;
+    return '|'  if -p $st;
+    return '='  if -S $st;
+    return '*'  if -x $st;      # must come after other tests
+    #return '%'                 # whiteout not supported
+    return '';
+}
+
+sub format_time {
+    my $time = shift;
+
+    my $fmt;
+    if ($time + $sixmonths > $curtime and $time < $curtime + $sixmonths) {
+        # dd mmm hh:mm
+        $fmt = "%b %e %R";
```

Comment:
Is there a not-too-complicated way of localizing date/times in Perl? Months are spelled out so it's not the end of the world, but I bet `ls` localizes dates.

Reply:
The month names are already localized automatically with the default format.

I'll add the -D option for strftime(3) date output, and -T for full time information with using long (-l) format.

Reply:
Ok, done.  This will go into the new PR's check-in.